components/Button.js

```diff
@@ -1,23 +1,55 @@
 import React, { PureComponent } from 'react';
-import { TouchableOpacity } from 'react-native';
+import { LayoutAnimation, TouchableOpacity } from 'react-native';
 import { connectAnimation } from '@shoutem/animation';
 import { connectStyle } from '@shoutem/theme';
+import { Spinner } from './Spinner';
 
 class Button extends PureComponent {
-  render() {
-    // The underlayColor is not a valid RN style
-    // property, so we have to unset it here.
-    const style = {
-      ...this.props.style,
+  componentDidUpdate(prevProps) {
+    const { inProgress: prevInProgress } = prevProps;
+    const { inProgress } = this.props;
+
+    if (prevInProgress !== inProgress) {
+      LayoutAnimation.easeInEaseOut();
```

Comment:
This won't take effect for the first change right ? First transition from not loading to loading